domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -12,24 +13,27 @@ namespace API.Controllers;
 [ApiController]
 public class TransferAgreementsController : ControllerBase
 {
-    private readonly ITransferAgreementService transferAgreementService;
+    private readonly ITransferAgreementRepository transferAgreementRepository;
 
-    public TransferAgreementsController(ITransferAgreementService transferAgreementService) => this.transferAgreementService = transferAgreementService;
+    public TransferAgreementsController(ITransferAgreementRepository transferAgreementRepository) => this.transferAgreementRepository = transferAgreementRepository;
 
     [HttpPost("api/transfer-agreements")]
     public async Task<ActionResult> Create([FromBody] CreateTransferAgreement request)
     {
+        var actor = User.FindActorClaim();
+        var subject = User.FindSubjectClaim();
+
         var transferAgreement = new TransferAgreement
         {
-            SenderId = Guid.Parse(User.FindFirstValue("sub")),
-            ActorId = User.FindFirstValue("atr"),
-            StartDate = request.StartDate.UtcDateTime,
-            EndDate = request.EndDate.UtcDateTime,
+            StartDate = DateTimeOffset.FromUnixTimeSeconds(request.StartDate),
+            EndDate = DateTimeOffset.FromUnixTimeSeconds(request.EndDate),
+            ActorId = actor,
+            SenderId = Guid.Parse(subject),
             ReceiverTin = request.ReceiverTin
```

Comment:
Is there a reason why there is no validation of the ReceiverTin?

Reply:
Because it isn't specified, in the issue, as a subtask. Implementing validations, should in my opinion be specified in a separate issue, due to the complexity, and additional tasks, it would add to the existing issue, regarding #1473, but also #1465.

I have made a list of validations that are potentially needed, as of this moment:

**1.** A TransferAgreement between the SAME Sender, and Receiver, cannot be created if the range between StartDate and EndDate overlap, with a previously created TransferAgreements start and enddate. For example you are not able to create another transferagreement if there is still a transfer agreement that is valid, since it hasn't expired yet. The start date of the new transferagreement must be set after the enddate of the previously created transferagreement.

**2.** A TransferAgreement cannot have the same subject as sender and receiver. Which means that the subject currently logged in can't send himself transfer agreements. The UUid for the currently logged in user, is collected from the user session jwt. it should be compared to the receiverTin's associated subjectId.

**3.** A transferAgreement cannot have a start date prior to the current UTC DatetimeOffset timestamp.

**4.** A transferagreement Cannot have an enddate that is prior or the same date as the startdate. The enddate must be a later date than the startdate.

**5.** A transferagreement MUST have a startdate, enddate?, ReceiverTin,

**6.** The ReceiverTin which the user sends in the creation Dto, must be 8 digits.


The amount of validations, combined with the complexity of implementing them, means that they should be refined, and specified in a separate issue.